File: CastIt.Domain/Enums/AppMessageType.cs
namespace CastIt.Domain.Enums
{
    public enum AppMessageType
    {
        UnknownErrorOccurred = 1,
        InvalidRequest = 2,
        NotFound = 3,

        PlayListNotFound = 100,

        UnknownErrorLoadingFile = 200,
        FileNotFound = 201,
        FileIsAlreadyBeingPlayed = 202,
        FileNotSupported = 203,
        FilesAreNotValid = 204,
        NoFilesToBeAdded = 205,
        UrlNotSupported = 206,
        UrlCouldntBeParsed = 207,
        OneOrMoreFilesAreNotReadyYet = 208,

        NoDevicesFound = 300,
        NoInternetConnection = 301,
        ConnectionToDeviceIsStillInProgress = 302,
    }
}

File: CastIt.Android/lib/domain/enums/app_message_type.dart
enum AppMessageType {
  unknownErrorOccurred,
  invalidRequest,
  notFound,
  playListNotFound,
  unknownErrorLoadingFile,
  fileNotFound,
  fileIsAlreadyBeingPlayed,
  fileNotSupported,
  filesAreNotValid,
  noFilesToBeAdded,
  urlNotSupported,
  urlCouldntBeParsed,
  oneOrMoreFilesAreNotReadyYet,
  noDevicesFound,
  noInternetConnection,
  connectionToDeviceIsStillInProgress,
  ffmpegError,
  serverIsClosing,
  ffmpegExecutableNotFound,
}

AppMessageType getAppMessageType(int value) {
  switch (value) {
    case 1:
      return AppMessageType.unknownErrorOccurred;
    case 2:
      return AppMessageType.invalidRequest;
    case 3:
      return AppMessageType.notFound;
    case 100:
      return AppMessageType.playListNotFound;
    case 200:
      return AppMessageType.unknownErrorLoadingFile;
    case 201:
      return AppMessageType.fileNotFound;
    case 202:
      return AppMessageType.fileIsAlreadyBeingPlayed;
    case 203:
      return AppMessageType.fileNotSupported;
    case 204:
      return AppMessageType.filesAreNotValid;
    case 205:
      return AppMessageType.noFilesToBeAdded;
    case 206:
      return AppMessageType.urlNotSupported;
    case 207:
      return AppMessageType.urlCouldntBeParsed;
    case 208:
      return AppMessageType.oneOrMoreFilesAreNotReadyYet;
    case 300:
      return AppMessageType.noDevicesFound;
    case 301:
      return AppMessageType.noInternetConnection;
    case 302:
      return AppMessageType.connectionToDeviceIsStillInProgress;
    case 303:
      return AppMessageType.ffmpegError;
    case 304:
      return AppMessageType.serverIsClosing;
    case 305:
      return AppMessageType.ffmpegExecutableNotFound;
  }
  throw Exception('The provided code = $value is not valid');
}
